src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -534,6 +568,16 @@ private void readUploadFile(BinaryData uploadFileBinary, UploadFileFilenameProce
         }
     }
 
+    private ArmResponseUploadFileRecord getResponseUploadFileRecordOrDelete(Path jsonPath) throws IOException {
```

Comment:
Just a thought, but given there are 3 very similar versions of this method, consider genericising it to avoid the duplication. I.e is there enough commonality for all 3 to be replaced with a single method with the following signature?
```java
private <T> T getRecordOrDelete(Path jsonPath, Class<T> clazz, String errorMessage) throws IOException {
```

Reply:
There is a ticket to refactor this class. It might be bests its done then.